toolprovider/mise/bootstrap.go

```diff
@@ -32,7 +33,45 @@ func installReleaseBinary(version string, targetDir string) error {
 		return fmt.Errorf("download %s: received status code %d", url, resp.StatusCode)
 	}
 
-	gzipReader, err := gzip.NewReader(resp.Body)
+	tempFile, err := os.CreateTemp("", "mise-*.tar.gz")
+	if err != nil {
+		return fmt.Errorf("create temp file: %w", err)
+	}
+	tempPath := tempFile.Name()
+	defer func() {
+		_ = tempFile.Close()
+		_ = os.Remove(tempPath)
+	}()
+
+	if _, err := io.Copy(tempFile, resp.Body); err != nil {
+		return fmt.Errorf("save download to temp file: %w", err)
+	}
+
+	if err := tempFile.Close(); err != nil {
+		return fmt.Errorf("close temp file: %w", err)
+	}
+
+	cmd := exec.Command("shasum", "-a", "256", tempPath)
```

Comment:
I'd prefer using the Go stdlib's SHA256 functions if possible. It's only a few lines of code and you can copy-paste the usage from here: https://github.com/bitrise-io/go-steputils/blob/master/cache/keytemplate/checksum.go#L86